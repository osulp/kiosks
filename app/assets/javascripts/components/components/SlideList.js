import React, { Component, PropTypes } from 'react';

class SlideList extends Component {

  render() {

    const { slides, setSlides } = this.props;

    const handleSubmit = (e) => {
      e.preventDefault();
    };

    const handleKeyUp = (e) => {
      console.log(`handleKeyUp, ${e.keyCode}`);
      if(e.keyCode == 13){
        if (typeof App !== 'undefined'){
          //App.room.speak(e.target.value);
        }else{
          setSlides();
        }
        e.target.value = "";
      };
    };

    return (
      <div>
        <ul>
          {slides.map((s) => {
            return <li key={`slide.${s.id}`}>{s.content}</li>;
          })
          }
        </ul>
        <form onSubmit={handleSubmit}>
          <input type="text" onKeyUp={handleKeyUp}/>
        </form>
      </div>
    );
  }
}

SlideList.propTypes = {
  slides: PropTypes.any,
};

export default SlideList;