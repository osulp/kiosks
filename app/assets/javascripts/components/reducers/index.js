import { combineReducers } from 'redux';
import slidesReducer from './slidesReducer';

const rootReducer = combineReducers({
  slides: slidesReducer
});

export default rootReducer;