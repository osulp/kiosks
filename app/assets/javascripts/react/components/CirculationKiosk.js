import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import Kiosk from './presentational/Circulation/Kiosk';
import * as actions from '../actions';

// Top level mapping of the application state to properties of component that is being
// connected.
export const mapStateToProps = (state) => {
  return {
    slides: state.kiosk.slides,
    url: state.kiosk.url,
    is_fetching_slides: state.touch.is_fetching_slides,
    show_nav: false,
    rooms_available_count: state.circ.rooms_available_count,
  }
};

// Map the actions to the dispatcher
export const mapDispatchToProps = (dispatch) => {
  return bindActionCreators(actions, dispatch);
};

// Connect the mappings (state -> properties, and actions -> dispatch) to the application component
export default connect(mapStateToProps, mapDispatchToProps)(Kiosk);
