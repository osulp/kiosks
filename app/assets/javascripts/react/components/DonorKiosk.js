import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';

import Kiosk from './presentational/Donor/Kiosk';
import * as actions from '../actions';

// Top level mapping of the application state to properties of component that is being
// connected.
export const mapStateToProps = (state) => {
  return {
    slides: state.kiosk.slides,
    title: state.kiosk.title,
    google_analytics: state.kiosk.google_analytics
  }
};

// Map the actions to the dispatcher
export const mapDispatchToProps = (dispatch) => {
  return bindActionCreators(actions, dispatch);
};

// Connect the mappings (state -> properties, and actions -> dispatch) to the application component
export default connect(mapStateToProps, mapDispatchToProps)(Kiosk);
