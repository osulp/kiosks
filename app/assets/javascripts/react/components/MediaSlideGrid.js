import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';

import MediaGrid from './presentational/Media/MediaGrid';
import * as actions from '../actions';

// Top level mapping of the application state to properties of component that is being
// connected.
export const mapStateToProps = (state) => {
  return {
    slides: state.kiosk.slides,
    title: state.kiosk.title,
    is_modal_visible: state.modal.visible
  }
};

// Map the actions to the dispatcher
export const mapDispatchToProps = (dispatch) => {
  return bindActionCreators(actions, dispatch);
};

// Connect the mappings (state -> properties, and actions -> dispatch) to the application component
export default connect(mapStateToProps, mapDispatchToProps)(MediaGrid);
