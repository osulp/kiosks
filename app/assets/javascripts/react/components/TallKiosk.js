import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import Kiosk from './presentational/Tall/Kiosk';
import * as actions from '../actions';

// Top level mapping of the application state to properties of component that is being
// connected.
export const mapStateToProps = (state) => {
  return {
    slides: state.kiosk.slides,
    restart_kiosk: state.kiosk.restart_kiosk,
    url: state.kiosk.url,
    is_fetching_slides: state.touch.is_fetching_slides,
    show_nav: state.circ.show_nav,
    rooms_available_count: state.circ.rooms_available_count,
    api: state.kiosk.api,
    hours: state.touch.hours,
    is_fetching_hours: state.touch.is_fetching_hours
  }
};

// Map the actions to the dispatcher
export const mapDispatchToProps = (dispatch) => {
  return bindActionCreators(actions, dispatch);
};

// Connect the mappings (state -> properties, and actions -> dispatch) to the application component
export default connect(mapStateToProps, mapDispatchToProps)(Kiosk);
