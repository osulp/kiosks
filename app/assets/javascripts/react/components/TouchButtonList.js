import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import ButtonList from './presentational/Touch/ButtonList';
import { fetchSlides } from '../actions/touchActions';

// Top level mapping of the application state to properties of component that is being
// connected.
const mapStateToProps = (state) => {
  return {
    url: state.kiosk.url
  }
};

// Map the actions to the dispatcher
const mapDispatchToProps = (dispatch) => {
  return {
    fetchSlides: fetchSlides
  }
};

// Connect the mappings (state -> properties, and actions -> dispatch) to the application component
export default connect(mapStateToProps, mapDispatchToProps)(ButtonList);
