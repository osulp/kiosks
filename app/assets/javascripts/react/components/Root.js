import React, {Component} from 'react';
import {render} from 'react-dom';
import {Provider} from 'react-redux';
import configureStore from '../store/configureStore';

import App from './App';
import TouchKiosk from './TouchKiosk';
import DonorKiosk from './DonorKiosk';
import CirculationKiosk from './CirculationKiosk';
import MediaKiosk from './MediaKiosk';
import InteractiveKiosk from './InteractiveKiosk';
import {setKiosk, setSlides, setRestartKiosk, setGoogleAnalytics} from '../actions/kioskActions';
import {setMaps} from '../actions/touchActions';

const store = configureStore();

export default class Root extends Component {

  /**
   * Prior to mounting, dispatch the Redux action to set the slides for the app
   */
  componentWillMount() {
    store.dispatch(setKiosk(this.props.kiosk_type, this.props.kiosk_url, this.props.kiosk_map_default_floor_number, this.props.kiosk_maps_base_url, this.props.kiosk_name, this.props.kiosk_id));
    store.dispatch(setSlides(this.props.slides));
    store.dispatch(setRestartKiosk(this.props.restart_kiosk));
    store.dispatch(setMaps(this.props.maps));
    if(typeof this.props.google_analytics != 'undefined') {
      store.dispatch(setGoogleAnalytics(this.props.google_analytics));
    }
  }

  /**
   * Render the provider to tie the store to the container React component
   * @returns {Component}
   */
  render() {
    return (
      <Provider store={store}>
        {this.getComponent(this.props.kiosk_type)}
      </Provider>
    );
  }

  /**
   * Given the type of kiosk to be rendered, return the proper application container
   * @param {string} kiosk_type - The type of container React component to render
   * @returns {Component}
   **/
  getComponent(kiosk_type) {
    switch(kiosk_type) {
      case 'touch':
        return (<TouchKiosk />);
      case 'donorcoverflow':
      case 'donor':
        return (<DonorKiosk />);
      case 'circulation':
        return (<CirculationKiosk />);
      case 'media':
        return (<MediaKiosk />);
      case 'interactive':
        return (<InteractiveKiosk />);
      default:
        return (<App />);
    }
  }
}
