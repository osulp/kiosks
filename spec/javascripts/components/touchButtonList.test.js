import { mapStateToProps, mapDispatchToProps } from '../../../app/assets/javascripts/react/components/TouchButtonList';
import * as factories from '../.factories';

describe('TouchButtonList', () => {
  it('maps state to props', () => {
    expect(mapStateToProps({kiosk: { url: "test" } })).toMatchSnapshot();
  });

  it('maps dispatch to props', () => {
    expect(mapDispatchToProps(jest.fn())).toMatchSnapshot();
  });
});
