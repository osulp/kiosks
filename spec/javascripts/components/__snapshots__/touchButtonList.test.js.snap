exports[`TouchButtonList maps dispatch to props 1`] = `
Object {
  "addError": [Function],
  "fetchSlides": [Function],
  "setKiosk": [Function],
  "setModalRootComponent": [Function],
  "setModalVisibility": [Function],
  "setSlides": [Function],
}
`;

exports[`TouchButtonList maps state to props 1`] = `
Object {
  "url": "test",
}
`;
